agent/src/main/java/com/teamscale/jacoco/agent/store/upload/UploadStoreBase.java

```diff
@@ -94,38 +92,38 @@ protected boolean tryUpload(String xml) {
 		} catch (IOException e) {
 			logger.error("Failed to upload coverage to {}. Probably a network problem", uploadUrl, e);
 			return false;
-		} catch (UploadStoreException e) {
+		} catch (UploaderException e) {
 			logger.error("Failed to upload coverage to {}. The configuration is probably incorrect", uploadUrl, e);
 			return false;
 		}
 	}
 
 	/**
-	 * Creates the zip file to upload which includes the given coverage XML and all
-	 * {@link #additionalMetaDataFiles}.
+	 * Creates the zip file to upload which includes the given coverage XML and all {@link #additionalMetaDataFiles}.
 	 */
-	private byte[] createZipFile(String xml) throws IOException {
-		try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
-			try (ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream);
-				 OutputStreamWriter writer = new OutputStreamWriter(zipOutputStream, "UTF-8")) {
-				fillZipFile(zipOutputStream, writer, xml);
+	private File createZipFile(File coverageFile) throws IOException {
+		File zipFile = createZipFileObject(coverageFile);
+		try (FileOutputStream fileOutputStream = new FileOutputStream(zipFile)) {
+			try (ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream)) {
+				fillZipFile(zipOutputStream, coverageFile);
 			}
-			return byteArrayOutputStream.toByteArray();
+			return zipFile;
 		}
 	}
 
+	private File createZipFileObject(File coverageFile) {
+		Path parentFolder = coverageFile.getParentFile().toPath();
+		String filename = coverageFile.getName().substring(0, coverageFile.getName().length() - 4) + ".zip";
```

Comment:
please use `FileSystemUtils.getFilenameWithoutExtension` instead as that conveys what's happening here better